web-registry/src/components/atoms/AccountLink.tsx

```diff
@@ -0,0 +1,48 @@
+import { ReactNode } from 'react';
+import { Link as RouterLink } from 'react-router-dom';
+import { Link as MuiLink, LinkProps as MuiLinkProps } from '@mui/material';
+import { truncate } from 'web-components/lib/utils/truncate';
+
+import { getAccountUrl } from '../../lib/block-explorer';
+
+interface AccountLinkProps extends MuiLinkProps {
+  address: string;
+  children?: ReactNode;
+}
+
+/** Takes an address and returns a link to either the ecocredits account page, or
+ * appropriate block explorer.
+ *
+ * @returns a `ReactRouterLink` for - Regen accounts, `MuiLink` for other addresses
+ *
+ * `children` are optional. If not passed, the address is truncated.
+ */
+export const AccountLink = ({
+  address,
+  children,
+  ...linkProps
+}: AccountLinkProps): JSX.Element => {
+  const isRegen = address.startsWith('regen');
+  const content = !!children ? children : truncate(address);
+  if (!isRegen) {
+    return (
+      <MuiLink
+        {...linkProps}
+        href={getAccountUrl(address)}
+        target={linkProps.target || '_blank'}
+        rel="noopener noreferrer"
```

Comment:
we might not need this in 2022, refs:
- https://stackoverflow.com/a/50709724/4028706
but then again maybe lighthouse scores are affected:
- https://web.dev/external-anchors-use-rel-noopener/

Reply:
Oh I didn't know this! interesting. 

Still seems like I might just add, given it doesn't have full browser support yet. Still nice to know, it's a bit of habit at this point